src/System.IO.Pipelines.Networking.Windows.RIO/Internal/Pool/MemoryPoolBlock.cs

```diff
@@ -134,13 +134,13 @@ protected override bool TryGetArray(out ArraySegment<byte> arraySegment)
             return true;
         }
 
-        public override MemoryHandle Pin()
+        public override MemoryHandle Pin(int byteOffset = 0)
         {
-            if (IsDisposed) RioPipelinesThrowHelper.ThrowObjectDisposedException(nameof(RioMemoryPoolBlock));
```

Comment:
Please note the removal of this check, since it was redundant with the check that Retain() already does (here and elsewhere).